src/makeInfinityAdder.js

```diff
@@ -4,7 +4,29 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  let i = 0;
+
+  const adder = (valueFirst) => {
+    if (i === 0 && !valueFirst) {
```

Comment:
The condition 'i === 0 && !valueFirst' is overly complex. You can simplify it by just checking '!valueFirst' since 'i' is not needed.